scripts/extensions/ai-widget/src/ai-assistant.tsx

```diff
@@ -0,0 +1,221 @@
+                                            }}
+                                            ariaValue={gettext('Close') + this.state.activeSection === 'headlines'
+                                                ? headlinesLabel : summaryLabel
+                                            }
+                                        />
+                                        <Heading type="h4" align="center">
+                                            {this.state.activeSection === 'headlines'
+                                                ? headlinesLabel : summaryLabel}
+                                        </Heading>
+                                    </Spacer>
+                                </div>
+                                <ContentDivider type="solid" margin="none" />
+                            </>
+                        )}
+                    </Spacer>
+                )}
+                body={(() => {
+                    if (this.state.activeSection == null) {
+                        return (
+                            <div
+                                className="
+                                    sd-grid-list
+                                    sd-grid-list--xx-small
+                                    sd-grid-list--gap-s
+                                    sd-grid-list--no-margin
+                                "
+                            >
+                                {configuration.generateHeadlines != null && (
+                                    <IllustrationButton
+                                        text={gettext('Headlines')}
+                                        onClick={() => {
+                                            this.setState({
+                                                activeSection: 'headlines',
+                                            });
+                                        }}
+                                    >
+                                        <SvgIconIllustration illustration="headlines" />
+                                    </IllustrationButton>
+                                )}
+                                {configuration.generateSummary != null && (
+                                    <IllustrationButton
+                                        text={gettext('Summary')}
+                                        onClick={() => {
+                                            this.setState({
+                                                activeSection: 'summary',
+                                            });
+                                        }}
+                                    >
+                                        <SvgIconIllustration illustration="summary" />
+                                    </IllustrationButton>
+                                )}
+                            </div>
+                        );
+                    } else if (this.state.activeSection === 'headlines') {
```

Comment:
Imagine a few more tabs got added, it would be a textbook definition of spaghetti code.

```
<header>
    {(() => {
        if (section === 'aaa') {
            // return ...
        } else if (section === 'bbb') {
            // return ...
        } else if (section === 'ccc') {
            // return ...
        } else if (section === 'ddd') {
            // return ...
        }
    })()}
</header>

<body>
    {(() => {
        if (section === 'aaa') {
            // return ...
        } else if (section === 'bbb') {
            // return ...
        } else if (section === 'ccc') {
            // return ...
        } else if (section === 'ddd') {
            // return ...
        }
    })()}
</body>

<footer>
    {(() => {
        if (section === 'aaa') {
            // return ...
        } else if (section === 'bbb') {
            // return ...
        } else if (section === 'ccc') {
            // return ...
        } else if (section === 'ddd') {
            // return ...
        }
    })()}
</footer>
```

Instead separate each tab into it's own component. A similar thing is implemented here https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/interactive-article-actions-panel/index-ui.tsx#L202

One difference in your case would be to persist state of closed tabs, but that's not too hard either - simply store state of all tabs in a root component.